transform-plugins/src/main/java/io/cdap/plugin/TransformFLLUtils.java

```diff
@@ -0,0 +1,122 @@
+  }
+
+  public static <C extends Collection<String>> C getOutputFields(StageSubmitterContext context,
+                                                                  Collector<String, ?, C> collector) {
+    Schema outputSchema = context.getInputSchema();
+    if (outputSchema == null || outputSchema.getFields() == null || outputSchema.getFields().isEmpty()) {
+      return null;
+    }
+
+    return outputSchema.getFields().stream().map(Schema.Field::getName).collect(collector);
+  }
+
+  public static void oneToOneIn(StageSubmitterContext context, String namePrefix, String descriptionPrefix) {
+    Set<String> input = getInputFields(context, Collectors.toSet());
+    if (input == null) {
+      return;
+    }
+
+    List<FieldOperation> operationList = new ArrayList<>();
+    for (String inputField : input) {
+      FieldTransformOperation operation =
+          new FieldTransformOperation(namePrefix + inputField, descriptionPrefix + " " + inputField,
+              Collections.singletonList(inputField), Collections.singletonList(inputField));
+      operationList.add(operation);
+    }
+    context.record(operationList);
+  }
+
+  public static void allInToFirstOut(StageSubmitterContext context, String name, String description) {
+    List<String> input = getInputFields(context, Collectors.toList());
+    List<String> output = getOutputFields(context, Collectors.toList());
+    if (input == null || output == null) {
+      return;
+    }
+
+    List<FieldOperation> operations = Collections
+        .singletonList(new FieldTransformOperation(name, description, input,
+                                                   Collections.singletonList(output.get(0))));
+    context.record(operations);
+  }
+
+  public static void firstInToAllOut(StageSubmitterContext context, String name, String description) {
+    List<String> input = getInputFields(context, Collectors.toList());
+    List<String> output = getOutputFields(context, Collectors.toList());
+    if (input == null || output == null) {
+      return;
+    }
+
+    List<FieldOperation> operations = Collections
+        .singletonList(new FieldTransformOperation(name, description, Collections.singletonList(input.get(0)),
+                                                   output));
+    context.record(operations);
+  }
+
+  public static void allInToAllOut(StageSubmitterContext context, String name, String description) {
+    List<String> input = getInputFields(context, Collectors.toList());
+    List<String> output = getOutputFields(context, Collectors.toList());
+    if (input == null || output == null) {
+      return;
+    }
+
+    List<FieldOperation> operations = Collections
+        .singletonList(new FieldTransformOperation(name, description, input, output));
+    context.record(operations);
+  }
+
+  public static void firstInToFirstOut(StageSubmitterContext context, String name, String description) {
```

Comment:
change this to `oneInToOneOut` so that it can record lineage for any one-to-one mapping